src/Exceptionless.Core/Services/StackService.cs

```diff
@@ -0,0 +1,113 @@
+            var occurrenceSetCacheKey = GetStackOccurrenceSetCacheKey();
+            var stackUsageSet = await _cache.GetSetAsync<Tuple<string, string, string>>(occurrenceSetCacheKey).AnyContext();
+            if (!stackUsageSet.HasValue || stackUsageSet.IsNull) return;
+            foreach (var tuple in stackUsageSet.Value) {
+                if (cancellationToken.IsCancellationRequested) break;
+
+                string organizationId = tuple.Item1, projectId = tuple.Item2, stackId = tuple.Item3;
+                string occurrenceCountCacheKey = GetStackOccurrenceCountCacheKey(organizationId, projectId, stackId),
+                    occurrenceMinDateCacheKey = GetStackOccurrenceMinDateCacheKey(organizationId, projectId, stackId),
+                    occurrenceMaxDateCacheKey = GetStackOccurrenceMaxDateCacheKey(organizationId, projectId, stackId);
+                var occurrenceCountTask = _cache.GetAsync<long>(occurrenceCountCacheKey, 0);
+                var occurrenceMinDateTask = _cache.GetAsync<double>(occurrenceMinDateCacheKey);
+                var occurrenceMaxDateTask = _cache.GetAsync<double>(occurrenceMaxDateCacheKey);
+
+                await Task.WhenAll(occurrenceCountTask, occurrenceMinDateTask, occurrenceMaxDateTask).AnyContext();
+                var occurrenceCount = (int)occurrenceCountTask.Result;
+                if (occurrenceCount == 0) continue;
+
+                await Task.WhenAll(
+                    _cache.DecrementAsync(occurrenceCountCacheKey, occurrenceCount, _expireTimeout),
+                    _cache.SetRemoveAsync(occurrenceSetCacheKey, tuple),
+                    _cache.RemoveAllAsync(new []{ occurrenceMinDateCacheKey, occurrenceMaxDateCacheKey })
+                ).AnyContext();
+
+                var occurrenceMinDate = occurrenceMinDateTask.Result.HasValue ? ConvertFromUnixTimeStamp(occurrenceMinDateTask.Result.Value) : SystemClock.UtcNow;
+                var occurrenceMaxDate = occurrenceMaxDateTask.Result.HasValue ? ConvertFromUnixTimeStamp(occurrenceMaxDateTask.Result.Value) : SystemClock.UtcNow;
+
+                try {
+                    if (!await _stackRepository.IncrementEventCounterAsync(organizationId, projectId, stackId, occurrenceMinDate, occurrenceMaxDate, occurrenceCount, sendNotifications).AnyContext()) {
+                        await IncrementStackUsage().AnyContext();
+                    }
+                    else {
+                        _logger.LogTrace("Increment event count {occurrenceCount} for organization:{organizationId} project:{projectId} stack:{stackId} with occurrenceMinDate:{occurrenceMinDate} occurrenceMaxDate:{occurrenceMaxDate}", occurrenceCount, organizationId, projectId, stackId, occurrenceMinDate, occurrenceMaxDate);
+                    }
+                }
+                catch {
+                    await IncrementStackUsage().AnyContext();
```

Comment:
We should add logging about what exception was thrown here, I'm also not sure if we should be incrementing the stack usage here. What if the previous IncrementStackUsage() is the one that threw, we could get a double count?

Reply:
Could you please help me to fix this issue?